api/admin/controller.py

```diff
@@ -157,11 +157,29 @@ def details(self, data_source, identifier):
         if isinstance(pool, ProblemDetail):
             return pool
         work = pool.work
+
         annotator = AdminAnnotator(self.circulation)
         return entry_response(
             AcquisitionFeed.single_entry(self._db, work, annotator)
         )
 
+    def edit(self, data_source, identifier):
+        """Edit a work's metadata."""
+
+        pool = self.load_licensepool(data_source, identifier)
+        if isinstance(pool, ProblemDetail):
+            return pool
+        work = pool.work
+        changed = False
+
+        new_title = flask.request.form.get("title")
+        if new_title and work.title != new_title:
+            work.primary_edition.title = new_title
+            changed = True
+        
+        if changed:
+            work.calculate_opds_entries()
```

Comment:
The old data will also be kept in the materialized views (at least until the views are refreshed) and the search index. I don't think it's worth doing anything about the materialized views, but you should update the work's entry in the search index.

This won't happen just from updating the title, but you may also need to call Work.calculate_presentation() after an edit operation.